select2.urs

```diff
@@ -1,5 +1,16 @@
-type t
-val create : (* options: *) xml [Cselect, Body] [] [] -> transaction t
-val createSingle (* only select one option *) : xml [Cselect, Body] [] [] -> transaction t
-val render : t -> xbody
-val selected : t -> signal (list string)
+con selector :: (Type -> Type) -> Type -> Type
+
+class singleOrMulti :: (Type -> Type) -> Type
+val singleOrMulti_option : singleOrMulti option
+val singleOrMulti_list : singleOrMulti list
+
+class mappable :: (Type -> Type) -> Type
+val mappable_option : mappable option
+val mappable_list : mappable list
+
+val createMulti : t ::: Type -> list (t * string * bool) -> transaction (selector list t)
+val createSingle : t ::: Type -> list (t * string) -> option int -> transaction (selector option t)
```

Comment:
It occurs to me that we don't need to get rid of `createSingle` in order to add
```
createRequiredSingle : t ::: Type -> list (t * string) -> option int -> transaction (selector ident t)
```
That itself would be pretty easy to write, and then all we'd need to do is add instances for `mappable_ident` and `singleOrMulti_ident`, which should be easy (although `singleOrMulti_ident` would be partial, and this would get in the way of my previous `IsList` idea since `ident` is definitely not list-like (to be fair, `toList` is fine and `fromList . toList` is still the identity, but it's not great that `fromList` is undefined on `[]`)).